pkg/graveler/ref/db_manager.go

```diff
@@ -762,13 +762,7 @@ func rWorker(ctx context.Context, d *pgxpool.Pool, rChan <-chan *graveler.Reposi
 	return nil
 }
 
-func Migrate(ctx context.Context, d *pgxpool.Pool, writer io.Writer) error {
-	cfg, err := config.NewConfig()
-	if err != nil {
-		return err
-	}
```

Comment:
Is the usage of `NewConfig` here is so bad, that we prefer to have the `cfg` as part of the `Migrate` prototype, and being ignored by almost all implementation?
On the other hand, maybe the config should be a singleton? I know it is out of the KV scope, but maybe we should open an issue for refactoring it into a singleton - WDYT?

Reply:
1. The usage here is bad because it introduce race conditions (the Migrate runs in parallel).
2. +1 on it being a singleton. I'll open the issue